library/src/main/java/com/novoda/downloadmanager/DownloadFile.java

```diff
@@ -47,9 +47,7 @@
     }
 
     void download(final Callback callback) {
-        callback.onUpdate(downloadFileStatus);
-
-        moveStatusToDownloadingIfQueued();
+        moveStatusToDownloadingIfQueuedOrError();
```

Comment:
When the `Batch.status` is Error it will be modified to `QUEUED` again, this is not the same for `File` which was just left as `ERROR`. Now we make sure that the `file` is updated. We could change this so that it is based off of the `DownloadBatch` instead, rather than just remapping.